### ISSUE TITLE
All vaults inaccessible after aborted photo upload

### ISSUE THREAD

**Issue by StanoRiga** [cryptomator/ios#55]
## Basic Info
I'm using iOS in version: 14.6
I'm running Cryptomator in version: 2.0.0 (438)

## Description
Hi all. I have a reproducible bug when it comes to upload photo files. Not sure if this also applies for other files as well (I guess so).
If you want to store a lot of photos at once in a cryptomator vault, this will lead to **all** vaults not accessible anymore.
What I did:

1. go to photo app.
2. select at least 150 photos
3. select "share -> safe in files"
4. select a cryptomator vault
5. select "save"  -> the loading circle in the right upper corner will appear and last forever. Wait for about half a minute. (first screenshot) (If you wait a long time there will pop up an error that a supplier app is not reacting. Waiting is optional for my scenario)
![IMG_5432 2021-07-11 06_56_25](https://user-images.githubusercontent.com/30843029/125186006-2b6c0200-e228-11eb-86f6-feb084a585bd.PNG)
6. select "cancel" in the left upper corner
7. now all vault are not longer accessible. if you click on them in files app, just nothing happens. If you click on them in the cryptomator app the files app will pop up and show a never ending loading screen. If you want to push files from other apps into a vault, the vaults are greyed out (2nd screenshot)
![IMG_5433 2021-07-11 06_56_26](https://user-images.githubusercontent.com/30843029/125186015-33c43d00-e228-11eb-95cb-b73ca71b500b.PNG)

To solve this you'll have to restart the phone.
Clean Log file (contains only entries where I did the described steps): [org.cryptomator.ios.fileprovider 2021-07-11--06-13-02-741.log](https://github.com/cryptomator/cryptomator-ios/files/6796507/org.cryptomator.ios.fileprovider.2021-07-11--06-13-02-741.log)

**Comment by tobihagemann** [comment]
All right, we've just released 2.0.0 Beta 8 and this time, we were actually able to reproduce and then fix the issue. We believe that the cause was a "thread explosion" that led to a crash of the File Provider. See 774b1cf74e7b0a1b247559cc064ad1633403e3c7.

Before closing this issue, it would be great if you can re-test this issue.

**Comment by StanoRiga** [comment]
Retested and wow… what a performance boost. 
1st I tried to reproduce the error with 150 photos, but that process was a matter of seconds, did not manage to abort it in time 😄. So result 1: large amount of photo upload does not last forever anymore. 
2nd I tested the abort process. Selected 300 photos/videos, share, safe to files, select vault, start process, cancel process. Abortion works as expected. I came back to the camera roll, also the selected files were still selected, and I was able to just start the process again. Did that a couple of times (starting and aborting) without any issues. 
And last but not least I extended Test one and selected over 500 photos/videos. Like a charm, also within a couple of seconds done. 
I checked the upload process in files app. Was going on smooth (not finished yet, but looks very good)

From my point of view this can be closed. 
Thank you!
